pyface/data_view/data_models/row_table_data_model.py

```diff
@@ -0,0 +1,126 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+""" A row-oriented data model implementation.
+
+This module provides a concrete implementation of a data model for the
+case of row-oriented data.
+"""
+from traits.api import Any, ComparisonMode, Instance, List, observe
+
+from pyface.data_view.abstract_data_model import (
+    AbstractDataModel, DataViewSetError
+)
+from pyface.data_view.index_manager import IntIndexManager
+from pyface.data_view.data_models.data_accessors import AbstractDataAccessor
+
+
+class RowTableDataModel(AbstractDataModel):
+    """ A data model that presents a list of objects as rows.
+    """
+
+    #: A sequence of objects to display in columns.
+    data = Any()
+
+    #: An object which describes how to map data for the row headers.
+    row_header_data = Instance(AbstractDataAccessor)
+
+    #: An object which describes how to map data for each column.
+    column_data = List(
+        Instance(AbstractDataAccessor),
+        comparison_mode=ComparisonMode.identity,
```

Comment:
Just a comment: This comparison mode setting won't be necessary on the next release of traits (if all goes to plan, that is.)